common/types/poet.go

```diff
@@ -126,49 +89,16 @@ func (p *PoetProofMessage) Ref() (PoetProofRef, error) {
 	return (PoetProofRef)(h), nil
 }
 
-type RoundEnd time.Time
-
-func (re RoundEnd) Equal(other RoundEnd) bool {
-	return (time.Time)(re).Equal((time.Time)(other))
-}
-
-func (re *RoundEnd) IntoTime() time.Time {
-	return (time.Time)(*re)
-}
-
-func (p *RoundEnd) EncodeScale(enc *scale.Encoder) (total int, err error) {
-	t := p.IntoTime()
-	n, err := scale.EncodeString(enc, t.Format(time.RFC3339Nano))
-	if err != nil {
-		return 0, err
-	}
-	return n, nil
-}
-
-// DecodeScale implements scale codec interface.
-func (p *RoundEnd) DecodeScale(dec *scale.Decoder) (total int, err error) {
-	field, n, err := scale.DecodeString(dec)
-	if err != nil {
-		return 0, err
-	}
-	t, err := time.Parse(time.RFC3339Nano, field)
-	if err != nil {
-		return n, err
-	}
-	*p = (RoundEnd)(t)
-	return n, nil
-}
-
 // PoetRound includes the PoET's round ID.
 type PoetRound struct {
 	ID  string `scale:"max=32"`
-	End RoundEnd
+	End time.Time
 }
 
 // ProcessingError is a type of error (implements the error interface) that is used to differentiate processing errors
 // from validation errors.
 type ProcessingError struct {
-	Err string `scale:"max=1024"` // TODO(mafa): make error code instead of string
```

Comment:
There seems to be only one type of processing error ("invalid poet id") maybe instead of a type we make it a value?

```golang
var ErrInvalidPoetID = errors.New("invalid poet id")

// ...

func (db *PoetDb) Validate(
	root []byte,
	proof types.PoetProof,
	poetID []byte,
	roundID string,
	signature types.EdSignature,
) error {
	const shortIDlth = 5 // check the length to prevent a panic in the errors
	if len(poetID) < shortIDlth {
		return fmt.Errof("%w: %x", ErrInvalidPoetID, poetID)
	}

	// ....
}
```

Reply:
Sure. I don't remember why we had it in the first place :shrug:

Reply:
Removed.